auto_nag/scripts/moved_to_performance.py

```diff
@@ -0,0 +1,145 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import json
+
+from auto_nag.bzcleaner import BzCleaner
+
+
+class MovedToPerformance(BzCleaner):
+    """Add a comment to bugs that recently moved to the performance component"""
+
+    def __init__(self):
+        super().__init__()
+        self.ni_extra = {}
+
+    def description(self):
+        return "Bugs that recently moved to the performance component"
+
+    def get_extra_for_needinfo_template(self):
+        return self.ni_extra
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        has_profiler_link = (
+            any(
+                "https://share.firefox.dev/" in comment["text"]
+                for comment in bug["comments"]
+            )
+            or "https://perf-html.io/from-url/" in bug["url"]
+        )
+
+        has_memory_report = any(
+            self._is_memory_report_file(attachment)
+            for attachment in bug["attachments"]
+            if not attachment["is_obsolete"] and not attachment["is_patch"]
+        )
+
+        has_troubleshooting_info = any(
+            self._is_troubleshooting_content(attachment)
+            for attachment in bug["attachments"]
+            if not attachment["is_obsolete"] and not attachment["is_patch"]
+        )
+
+        if has_profiler_link and has_memory_report and has_troubleshooting_info:
+            # Nothing missing, no need to add a comment
+            return None
+
+        self.ni_extra[bugid] = {
+            "has_profiler_link": has_profiler_link,
+            "has_memory_report": has_memory_report,
+            "has_troubleshooting_info": has_troubleshooting_info,
+        }
+
+        return bug
+
+    @staticmethod
+    def _is_memory_report_file(attachment):
+        return (
+            attachment["content_type"]
+            in (
+                "application/gzip",
+                "application/x-gzip",
+                "application/json",
+            )
+            and "memory-report" in attachment["file_name"]
+        )
+
+    @staticmethod
+    def _is_troubleshooting_content(attachment):
+        if (
+            attachment["content_type"]
+            not in (
+                "application/json",
+                "text/plain",
+            )
+            or "memory-report" in attachment["file_name"]
+        ):
+            return False
+
+        content_bytes = base64.b64decode(attachment["data"])
+        try:
+            content = json.loads(content_bytes)
```

Comment:
Sometimes people attach textual info, which looks like:
```
Informazioni di base
--------------------

Nome: Firefox
Versione: 112.0a1
ID build: 20230312211644
...

Segnalazioni di arresto anomalo degli ultimi 3 giorni
-----------------------------------------------------
...
```

Not sure if we can easily detect it, as it's translated.
Maybe we could just check if there are multiple lines with multiple `-` characters.

Reply:
The comment will require clinking on "Copy **raw** data to clipboard", not "Copy **text** to clipboard". So, it should be fine if we ask for the data if it is not matching the expected format.

WDYT?

Reply:
Yes this works for me. If we see this brings problems we can adjust later.